webapp/shop/api/ua_contracts/models.py

```diff
@@ -133,6 +146,16 @@ def __init__(
         self.created_at = created_at
         self.actionable = actionable
         self.discount = discount
+        # Properties below apply only to channel offers.
+        # If activation_account_id exist, it's a channel offer.
+        if activation_account_id is not None:
+            self.can_change_items = can_change_items
+            self.external_ids = external_ids
+            self.activation_account_id = activation_account_id
+            self.distributor_account_name = distributor_account_name
+            self.reseller_account_name = reseller_account_name
+            self.end_user_account_name = end_user_account_name
```

Comment:
@jpmartinspt  According to [ua_contract doc](https://github.com/canonical/ua-contracts/blob/d32984c23db25bb09858ca59b939cb2dbb3d493f/docs/contracts.yaml#L2614-L2721), `externalIDs` and `canChangeItems` are required properties.  Theses two properties are not used for existing offers but used for the channel offer. `activation_account_id` is the property to identify a channel offer.  To avoid making any changes to existing offers, can I reference these attributes only when it's a channel offer (when `activation_account_id` is not `None`) ?

Reply:
Yes, I think that is fine for now. maybe we can had a method in the Offer class that returns if an offer is a channel offer or not and unit test that logic too?